src/augmented_block.jl

```diff
@@ -121,6 +121,36 @@ function get_hessian(m::AugmentedNLPBlockModel;
     return H
 end
 
+"""
```

Comment:
that's exactly the idea.
I would just suggest a change in the name (for me, the primal and dual variables in the submodel are not solution, but parameters). How about just using `update_primal!` and `update_dual!`?